### PR TITLE
feat: if print mode hide unnecessary empty space

### PR DESCRIPTION
**Pull request by ktx-vaidehi**
- #4870

## Summary by CodeRabbit

- **New Features**
	- Enhanced table rendering for print mode, ensuring proper styling and positioning.
	- Introduced conditional rendering for dashboard panels specifically in print scenarios, improving layout adaptability.

- **Bug Fixes**
	- Improved responsiveness of the dashboard display based on the application's print state.

### CODE REVIEW

**Review by ktx-kirtan — APPROVED** [review]

File: web/src/components/dashboards/panels/TableRenderer.vue [hunk 1]
```diff
@@ -43,6 +46,7 @@ import useNotifications from "@/composables/useNotifications";
 import { exportFile } from "quasar";
 import { defineComponent, ref } from "vue";
 import { findFirstValidMappedValue } from "@/utils/dashboard/convertDataIntoUnitValue";
+import { useStore } from "vuex";
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/panels/TableRenderer.vue:49]: this is not required

File: web/src/views/Dashboards/RenderDashboardCharts.vue [hunk 2]
```diff
@@ -42,7 +42,39 @@
...
+          :searchType="searchType"
+          @updated:data-zoom="$emit('updated:data-zoom', $event)"
+          @onMovePanel="onMovePanel"
+          @refreshPanelRequest="refreshPanelRequest"
+          @refresh="refreshDashboard"
+          @update:initial-variable-values="updateInitialVariableValues"
+          @onEditLayout="openEditLayout"
+          style="height: 100%; width: 100%"
+        />
+      </div>
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/views/Dashboards/RenderDashboardCharts.vue]: done this changes

File: web/src/views/Dashboards/RenderDashboardCharts.vue [hunk 3]
```diff
@@ -42,7 +42,39 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+        <PanelContainer
+          @onDeletePanel="onDeletePanel"
+          @onViewPanel="onViewPanel"
+          :viewOnly="viewOnly"
+          :data="panels[0]"
+          :dashboardId="dashboardData.dashboardId"
+          :folderId="folderId"
+          :selectedTimeDate="
+            currentTimeObj[panels[0].id] || currentTimeObj['__global'] || {}
+          "
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/views/Dashboards/RenderDashboardCharts.vue]: it is fixed

File: web/src/views/Dashboards/RenderDashboardCharts.vue [hunk 4]
```diff
@@ -42,7 +42,39 @@
...
+          @onMovePanel="onMovePanel"
+          @refreshPanelRequest="refreshPanelRequest"
+          @refresh="refreshDashboard"
+          @update:initial-variable-values="updateInitialVariableValues"
+          @onEditLayout="openEditLayout"
+          style="height: 100%; width: 100%"
+        />
+      </div>
       <grid-layout
+        v-else
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/views/Dashboards/RenderDashboardCharts.vue:78]: this can create issue in other charts

**Pull request merged**